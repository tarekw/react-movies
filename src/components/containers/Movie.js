import Movie from '../ui/Movie'
import { connect } from 'react-redux'

const mapStateToProps = (state, props) => ({
    movieDetails: state.movieDetails,
});

export default connect(mapStateToProps)(Movie);
