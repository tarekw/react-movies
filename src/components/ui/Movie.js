import React from 'react';
import PropTypes from 'prop-types';
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import Rating from '@material-ui/lab/Rating';

const IMAGE_BASE = 'https://image.tmdb.org/t/p/w300/';

const Movie = ({ movieDetails={} }) => (
    <React.Fragment>
        <CardMedia
            style={{height: 450, width: 300}}
            image={IMAGE_BASE+movieDetails.poster_path}
        />
        <Rating name="half-rating" value={(movieDetails.vote_average / 10) * 5} precision={0.5} />
        <Typography gutterBottom variant="headline" component="h2">
            Title: {movieDetails.title}
        </Typography>
        <Typography gutterBottom variant="headline" component="h2">
            Description: {movieDetails.overview}
        </Typography>
    </React.Fragment>
);

Movie.propTypes = {
    movieDetails: PropTypes.object
}

export default Movie
