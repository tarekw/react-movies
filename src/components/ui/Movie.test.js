import React from 'react';
import { shallow } from 'enzyme';

import Movie from './Movie';

const mockMovie = {
	"id": 475557,
	"poster_path": "/udDclJoHjfjb8Ekgsd4FDteOkCU.jpg",
	"title": "Joker",
	"overview": 'overview',
	"vote_average": 8.7,
};

it('should render correctly with default props', () => {
    const component = shallow(<Movie />);
    expect(component).toMatchSnapshot();
});

it('should render correctly with mocked props', () => {
    const component = shallow(<Movie movieDetails={mockMovie} />);
    expect(component).toMatchSnapshot();
});
