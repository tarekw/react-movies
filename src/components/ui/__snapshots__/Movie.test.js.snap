// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with default props 1`] = `
<Fragment>
  <WithStyles(ForwardRef(CardMedia))
    image="https://image.tmdb.org/t/p/w300/undefined"
    style={
      Object {
        "height": 450,
        "width": 300,
      }
    }
  />
  <WithStyles(ForwardRef(Rating))
    name="half-rating"
    precision={0.5}
    value={NaN}
  />
  <WithStyles(ForwardRef(Typography))
    component="h2"
    gutterBottom={true}
    variant="headline"
  >
    Title: 
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    component="h2"
    gutterBottom={true}
    variant="headline"
  >
    Description: 
  </WithStyles(ForwardRef(Typography))>
</Fragment>
`;

exports[`should render correctly with mocked props 1`] = `
<Fragment>
  <WithStyles(ForwardRef(CardMedia))
    image="https://image.tmdb.org/t/p/w300//udDclJoHjfjb8Ekgsd4FDteOkCU.jpg"
    style={
      Object {
        "height": 450,
        "width": 300,
      }
    }
  />
  <WithStyles(ForwardRef(Rating))
    name="half-rating"
    precision={0.5}
    value={4.35}
  />
  <WithStyles(ForwardRef(Typography))
    component="h2"
    gutterBottom={true}
    variant="headline"
  >
    Title: 
    Joker
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    component="h2"
    gutterBottom={true}
    variant="headline"
  >
    Description: 
    overview
  </WithStyles(ForwardRef(Typography))>
</Fragment>
`;
